base/markdown/GitHub/table.jl

```diff
@@ -0,0 +1,168 @@
+                        else
+                            # default is align right
+                            push!(align, :r)
+                        end
+                    end
+                end
+
+            elseif n == 1 || length(rows[1]) == length(row)
+                push!(rows, map(x -> parseinline(x, config), row))
+            elseif length(row) > 1
+                seek(stream, pos)
+                break
+            else
+                return false
+            end
+        end
+        if length(rows) < 2
+            return false
+        end
+        push!(md, Table(rows, align))
+        return true
+    end
+end
+
+function html(io::IO, md::Table)
+    withtag(io, :table) do
+        for (i, row) in enumerate(md.rows)
+            withtag(io, :tr) do
+                for c in md.rows[i]
+                    t = (i == 1) ? :th : :td
+                    withtag(io, t) do
+                        htmlinline(io, c)
+                    end
+                end
+            end
+        end
+    end
+end
+
+function plain(io::IO, md::Table)
+    plain_cells = map(row -> map(Markdown.plaininline, row), md.rows)
+    plain_lengths = map(row -> int(map(length, row)), plain_cells)
+    col_widths = reduce(max, plain_lengths)
+
+    for i in 1:length(md.rows)
+        for (j, text) in enumerate(plain_cells[i])
+            (j != 1) && print(io, "  | ")
+            print(io, " " ^ (col_widths[j] - plain_lengths[i][j]))
+            print(io, text)
+        end
+        println(io)
+
+        if i == 1
+            for (j, w) in enumerate(col_widths)
+                if j != 1
+                    print(io, "  | ")
+                end
+                a = typeof(md.align) == Symbol ? md.align : md.align[j]
+                print(io, _dash(w, a))
+            end
+            println(io)
+        end
+    end
+end
+
+function _dash(width, align)
+    if align == :l
+        return ":" * "-" ^ max(3, width - 1)
+    elseif align == :r
+        return "-" ^ max(3, width - 1) * ":"
+    elseif align == :c
+        return ":" * "-" ^ max(3, width - 2) * ":"
+    else
+        throw(ArgumentError("Unrecognized alignment $align"))
+    end
+end
+
+
+function writemime(io::IO, ::MIME"text/latex", md::Table)
+    wrapblock(io, "tabular") do
+        if typeof(md.align) == Symbol
+            align = string(md.align) ^ length(md.rows[1])
+        else
+            align = md.align
+        end
+        println(io, "{$(join(align, " | "))}")
+        for (i, row) in enumerate(md.rows)
+            for (j, cell) in enumerate(row)
+                if j != 1
+                    print(io, " & ")
+                end
+                latex_inline(io, cell)
+            end
+            println(io, " \\\\")
+            if i == 1
+                println("\\hline")
+            end
+        end
+    end
+end
+
+function term(io::IO, md::Table, columns)
+    col_widths = reduce(max, map(cell -> int(map(ansi_length, cell)),
```

Comment:
the problem is that this is not equivalent, plain includes things like `*` or `[]()`, similarly if we were to capture the print it would include ansi codes.

I think we can refactor after merging... :)